### PR TITLE
Document the pytest_xdist_auto_num_workers hook

### PR DESCRIPTION
**Pull request by encukou**
Document the pytest_xdist_auto_num_workers hook

I also reorganized the `-n` section slightly, attempting to put the most generally useful information first, and dive into details later.

The feature and tests were added in #590.

### CODE REVIEW

File: docs/distribution.rst [hunk 1]
```diff
@@ -5,12 +5,24 @@ Running tests across multiple CPUs
...
 
+With ``-n auto``, pytest-xdist will use as many processes as your computer
+has CPU cores.
+Pass a number, e.g. ``-n 8``, to specify the number of processes explicitly.
+
+To specify a different meaning for ``-n auto`` for your tests,
+you can implement the ``pytest_xdist_auto_num_workers``
+`pytest hook <https://docs.pytest.org/en/latest/how-to/writing_plugins.html>`__
+(a function named ``pytest_xdist_auto_num_workers`` in e.g. ``conftest.py``)
+that returns the number of workers ``-n auto``.
```
Comment (nicoddemus) [inline docs/distribution.rst:21]: Not sure what you mean by `-n auto` at the end of this sentence... or did you mean to end with `returns the number of workers.`?

**Comment by nicoddemus** [comment]
Looks good, except my note below.

File: docs/distribution.rst (hunk 1 above)
Comment (encukou) [reply docs/distribution.rst:21]: Thanks for the catch!
```suggestion
that returns the number of processes to use.
```

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt is py36 meant to fail due to setuptools-scm?

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus damn that setuptools bug, setuptools_scm dopped that dead snake, i'd say merge is ok, i will debug later hopefully

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus i think i have a idea

**Comment by RonnyPfannschmidt** [comment]
#793 fixes py36

**Comment by nicoddemus** [comment]
Thanks @RonnyPfannschmidt!

**Pull request merged**